File: sammy/230425_study/프로_76502_괄호 회전하기.py
def solution(s):
    answer = 0
    check=s

    for _ in range(len(s)):
        # 괄호 하나씩 뒤에 붙이기
        check_=check[1:]+check[0]
        # 변경시킨 check_를 check에 반영
        check=check_
        
        for _ in range(len(s)//2): # s 길이의 2로 나눈만큼 (),{},[] replace 진행하여 ""로 만들어주기
            check_=check_.replace('()','')
            check_=check_.replace('{}','')
            check_=check_.replace('[]','')

        if check_=="": # "" 갯수 카운팅
            answer+=1

    return answer

print(solution("[](){}"),3)
print(solution("}]()[{"),2)
print(solution(	"[)(]"),0)
print(solution("}}}"),0)

File: James/230425_study/프로_76502_괄호회전하기.java
import java.util.*;

class Solution {
    public boolean isCorrect(String newS) {
        Stack<Character> stack = new Stack<>();
        
        try {
            for(int i = 0; i < newS.length(); i++) {
                char bracket = newS.charAt(i);
            
                switch(bracket) {
                    case '(' :
                        stack.add(bracket);
                        break;
                    
                    case ')' :
                        if(stack.peek() != '(')
                            return false;
                    
                        stack.pop();
                        break;
                    
                    case '{' :
                        stack.add(bracket);
                        break;
                    
                    case '}' :
                        if(stack.peek() != '{')
                            return false;
                    
                        stack.pop();
                        break;
                    
                    case '[' :
                        stack.add(bracket);
                        break;
                    
                    case ']' :
                        if(stack.peek() != '[')
                            return false;
                    
                        stack.pop();
                        break;
                    }
                }    
        } catch(Exception e) {
            return false;
        }
        
        return stack.isEmpty() ? true : false;
    }

    public int solution(String s) {
        int answer = 0;
        String newS;
        
        for(int i = 0; i < s.length(); i++) {
            String firstS = s.substring(0, i);
            String secondS = s.substring(i, s.length());
            
            newS = secondS + firstS;
            
            if(isCorrect(newS))
                answer++;            
        }
            
        return answer;
    }
}
